### DISCUSSION TITLE
Easiest way to recreate the GitHub online copy of a repo from local clone?

### DISCUSSION THREAD

**Discussion by OneOfMany07** [community/community#37437] [Other Feature Feedback, Questions, & Ideas] [answered]
TL;DR - I have a local git clone that thinks it has a GitHub remote repo, but I deleted it and wanted to recreate it the easiest way possible.

I don't use GitHub much, nor even git much lately.  I was trying to remove a personal fork of a now old parent (had been moved to a new location on GitHub with new owners, etc).  And I accidentally deleted a related project's online copy first instead.  I realized my mistake and deleted the fork, but now wonder what is the easiest way to recreate the online copy from my local clone?

Do I need to go through the process of creating a new repo?  And just fill in the name to match then create a new remote on the local repo to replace origin?

Or is there an easy way using the GitHub Desktop app?  I tried just clicking "Push origin" but it errored out that the remote repo didn't exist (OK, expected...but I'd almost hoped the app would prompt me to recreate it).

**Bonus detail:**  If I happen to want to change the name (since I'm starting over) does that change much?  I've never tried to rename a git repo before...

I haven't done a lot of git stuff for years now, and even back when it was my daily thing I never felt like it was invisible to me.  I often had to memorize the magic command for a purpose on anything beyond fetch/commit/push to origin.  Or look them up again... hadn't used GitHub Desktop back then (was a company project, this is personal work).

**Comment by antonblr** [comment, marked as answer]
@OneOfMany07 - you're correct. To restore it:
1) Goto GitHub and create a repo with the same name
2) Run `git push --all origin` to push your local repo to remote

Or, if you want to change the name:

1) Goto GitHub and create a repo with any name
2) In the local repo run `git remote set-url origin git@github.com:OneOfMany07/<NEW-NAME>.git` to update the remote url
3) Run `git push --all origin` to push your local repo to the new remote (feel free to rename the local directory as well)
